File: nomad-samples/admin/js/basic-live-channel/live-operator/cancel-segment.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function cancelSegment(AUTH_TOKEN, ID) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
    HEADERS.append("Content-Type", "application/json");

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/admin/liveOperator/${ID}/cancelSegment`, {
        method: "POST",
        headers: HEADERS
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {
        return await RESPONSE.json();
    }

    await apiExceptionHandler(RESPONSE, `Canceling Segment ${ID} failed`);
}

File: nomad-samples/admin/py/basic-live-channel-1.1/live_operator/cancel_segment.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import requests

def cancel_segment(AUTH_TOKEN, ID):
    
    HEADERS = {
        "Authorization": "Bearer " +  AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    try:
        RESPONSE = requests.post(f"{ADMIN_URL}/admin/liveOperator/{ID}/cancelSegment", headers= HEADERS)

        # If not found return None
        if (not RESPONSE.ok):
            raise Exception()
    
    except:
        api_exception_handler(RESPONSE, f"Cancelling segment for Live Channel {ID} failed")